addons/mail/models/res_users.py

```diff
@@ -144,6 +144,11 @@ def write(self, vals):
                 )
         return write_res
 
+    def action_archive(self):
```

Comment:
I've written a test as requested, but I am not sure if it's the correct way to test it. I am not really certain how I should test mail_activity, because it's a mixin. For now I rely on `post_install` to use a dummy `crm.lead` on which I can schedule activities. There were no prior tests besides a tour for mail_activity in the `mail` module.

Let me know if there's a better to go about it.

Reply:
You may have a look at `test_mail`module which holds globally most tests related to mail. It has models used for testing, so that we don't rely on existing models for testing purpose.

Reply:
Thanks for pointing this out, I've moved and adapted the test into the `test_mail` module.

Let me know what you think about it.